scribble-lib/scribble/blueboxes.rkt

```diff
@@ -135,8 +175,10 @@
                (error 'build-blueboxes-cache
                       "blueboxes info didn't have the right shape: ~s"
                       candidate))
-             candidate))
-         (set-bluebox-info-offset! a-bluebox-info (and desed (+ (string->number first-line) pos)))
+             (cast candidate Blueboxes-Info-Hash)))
+         (when (and desed (string? first-line))
+           (define first-line-num (assert (string->number first-line) exact-nonnegative-integer?))
```

Comment:
I think that probably this assert can actually fail (if the file is corrupt or serialization changes somehow or something) and so it should go thru the same path as line 169 (in the new code) where we log a warning.